src/main/java/uk/ac/ebi/tsc/portal/api/team/controller/TeamRestController.java

```diff
@@ -128,20 +128,16 @@ public TeamRestController(TeamService teamService,
 	public Resources<TeamResource> getAllTeamsForCurrentUser(Principal principal){
 
 		Collection<Team> teams = teamService.findByAccountUsername(principal.getName());
-		return new Resources<>(teams.stream().map(
-				TeamResource::new
-				).collect(Collectors.toList())
-				);
+		Collection<TeamResource> teamResources = teamService.populateTeamContactEmails(teams, principal);
```

Comment:
That would be a performance killer to get contact emails for each team one by one, right? Besides, I am not sure, where this method is used and if we need team emails here at all.

Reply:
Yes agree, except when a specific team is retrieved, contact emails need not be returned here. Will remove that.

Reply:
This is done.

Reply:
Where? I can still see unchanged code.